cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/DefaultApplicationUpdateContext.java

```diff
@@ -148,10 +160,10 @@ public String getConfigAsString() {
                                       pluginType, pluginName, predicate, limit, ArtifactSortOrder.ASC);
       for (Map.Entry<ArtifactDescriptor, PluginClass> pluginsEntry : plugins.entrySet()) {
         ArtifactId plugin = pluginsEntry.getKey().getArtifactId();
-        // Only consider non-SNAPSHOT plugins for upgrade.
-        // TODO: Consider making this check optional. Helpful for integration tests.
         if (!plugin.getVersion().isSnapshot()) {
           pluginArtifacts.add(plugin);
+        } else if (plugin.getVersion().isSnapshot() && considerSnapshotArtifacts) {
+            pluginArtifacts.add(plugin);
```

Comment:
Wrong alignment.

Reply:
Also, seems like the conditions can be combined. The IntelliJ IDE probably already suggesting it?